aks_cluster.tf

```diff
@@ -1,19 +1,25 @@
+data "azurerm_kubernetes_service_versions" "current" {
+  location = var.resource_group_location
+}
+
 resource "azurerm_kubernetes_cluster" "k8s" {
   name                = "${var.cluster_name}-cluster"
-  location            = var.resource_group_name
-  resource_group_name = var.resource_group_location
+  location            = var.resource_group_location
+  resource_group_name = var.resource_group_name
   dns_prefix          = "k8scluster"
-  kubernetes_version  = var.kubernetes_version
+  kubernetes_version  = data.azurerm_kubernetes_service_versions.current.latest_version
 
   default_node_pool {
-    name                = var.node_pool_name
-    node_count          = var.node_pool_count
-    vm_size             = var.node_pool_vm_size
-    type                = "VirtualMachineScaleSets"
-    os_disk_size_gb     = var.node_pool_osdisk_size
-    enable_auto_scaling = true
-    max_count           = var.node_pool_max_count
-    min_count           = var.node_pool_min_count
+    availability_zones   = [1]
```

Comment:
whats this variable?

Reply:
@KennedyUC

Reply:
Allows for AKS nodes to run across Availability Zones. Forms a protection against any single failure of a datacenter. Just used 1 for testing the k8s cluster.